bitnami/influxdb/templates/influxdb/configmap-backup.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if .Values.backup.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "influxdb.fullname" . }}-backup
+  labels: 
+    {{- include "influxdb.labels" . | nindent 4 }}
+    app.kubernetes.io/component: influxdb
+data:
+  backup.sh: |-
+    #!/bin/bash
+
+    set -e
+
+    DATE="$(date +%Y%m%d_%H%M%S)"
+
+    echo "deleting old backups"
+    find {{ .Values.backup.directory }}  -mindepth 2 -maxdepth 2 -type d -mtime +{{ .Values.backup.retentionDays }} -exec rm -r {} \;
```

Comment:
Shouldn't you remove old backups once you complete the new one? Imagine the backup fails (for any reason), you'll lose everything

Reply:
i had storage space in mind. 
free space by deleting the last one before creating the new one.

Reply:
IMHO it's better to double the space (for a few seconds) in the volume you attach to the cronjob pods, rather than losing data. 

Let's imagine this scenario:

- 1st backup creation works
- Sth fail while uploading it to GCE/Azure

Next time the cronjob is executed, it removes the 1st backup (which was never uploaded)